dimscord/objects/typedefs.nim

```diff
@@ -20,13 +23,14 @@ type
         events*: Events
         token*: string
         shards*: Table[int, Shard]
+        waits*: WaitTable

```

Comment:
Wondering if for ease of implementing the helper procs, we put this in the `RestApi`
Doesn't make as much sense but means that we can then have helpers like
```nim
await someMessage.waitForDeletion()
```

Reply:
Well shouldn’t Shard have the waits field? Messages are received from different shards. Every event always contain shard. DiscordClient is more general.

Reply:
I thought of making stand alone `RestApi` when I was making the library and DiscordClient would just be general. Since dimscord also supports rest only. Other libs like nostrum, discljord, etc I think do have that idea of `client.http/api`. I prolly could've just went with `DiscordClient` instead of `RestApi`, but I think it would be a breaking change having to migrate it. I'd just keep it as it is.